directives/thebe.py

```diff
@@ -277,6 +277,54 @@ def run(self):
         return [ThebeButtonNode(**kwargs)]
 
 
+class ThebePrecellButtonNode(nodes.Element):
+    """Appended to the doctree by the ThebeButton directive
+
+    Renders a nice looking activation button when placed before code cell
+    """
+
+    def __init__(self, rawsource="", *children, text="Activate", **attributes):
+        super().__init__("", text=text)
```

Comment:
Is it intentional that `*children` and `**attributes` are not passed to `super().__init__(...)`? Now those parameters are lost if the caller used them.

This is what I thought:
`super().__init__("", *children, text=text, **attributes)`

Reply:
This code came directly from upstream, so it's not intentional on my part. I'm not super familiar with sphinx, so am not sure if the parameters might be needed by a user at a later point.

Reply:
you mean the Thebelab upstream? It's fine, we can leave it as it is especially if the code really originates from elsewhere. Is there any docstring in the code that mentions the original source?

Reply:
In the [sphinx-thebe upstream](https://github.com/executablebooks/sphinx-thebe/blob/aec1ec48a98ceac0acaad4d5ac23e86a6047498b/sphinx_thebe/__init__.py#L185-L186). That file is mentioned at the top of `thebe.py`. To be more precise I copy-pasted that code from what the upstream had for `ThebeButtonNode` (since the two directives and nodes are nearly identical).